pbrd/pbr_vty.c

```diff
@@ -444,12 +444,37 @@ DEFPY(pbr_map_vrf, pbr_map_vrf_cmd,
 		goto done;
 	}
 
-	if (pbrms->vrf_lookup || pbrms->vrf_unchanged) {
-		vty_out(vty, SET_VRF_EXISTS_STR);
-		ret = CMD_WARNING_CONFIG_FAILED;
+	/*
+	 * Determine if a set vrf * command already exists.
+	 *
+	 * If its equivalent, just return success.
+	 *
+	 * Else, return failure, we don't allow atomic swaps yet.
+	 */
+	if (vrf_name && pbrms->vrf_lookup) {
+		/* New vrf specified and one already exists */
+
+		/* Is this vrf different from one already configured? */
+		if (strncmp(pbrms->vrf_name, vrf_name, sizeof(pbrms->vrf_name))
+		    != 0)
+			goto vrf_exists;
+
+		goto done;
```

Comment:
Can we replace goto done with return <whatever you want to return>? all over the function.
just oneline block doesn't need label. From readability-wise look good and easy to understand.

Reply:
Agreed, its easier to read with returns. Changed.